src/ed25519.rs

```diff
@@ -727,36 +719,27 @@ impl PublicKey {
     ///
     /// # Return
     ///
-    /// Returns true if the signature was successfully verified, and
-    /// false otherwise.
-    pub fn verify<D>(&self, message: &[u8], signature: &Signature) -> bool
+    /// Returns `Ok(())` if the signature is valid, and `Err` otherwise.
+    #[allow(non_snake_case)]
+    pub fn verify<D>(&self, message: &[u8], signature: &Signature) -> Result<(), SignatureError>
             where D: Digest<OutputSize = U64> + Default
     {
-        let mut h: D = D::default();
-        let mut a: EdwardsPoint;
-        let ao:  Option<EdwardsPoint>;
-        let mut digest: [u8; 64] = [0u8; 64];
-
-        ao = self.decompress();
-
-        if ao.is_some() {
-            a = ao.unwrap();
-        } else {
-            return false;
-        }
-        a = -(&a);
+        let A = self.0.decompress()
+            .ok_or_else(|| SignatureError(InternalError::PointDecompressionError))?;
```

Comment:
I think I prefer the way I did it with `match` in #31? If that's okay.

Reply:
It's up to you, but I think this is a lot cleaner.